Detections/AzureDevOps/AzDOHistoricPrPolicyBypassing.yaml

```diff
@@ -0,0 +1,44 @@
+id: 4d8de9e6-263e-4845-8618-cd23a4f58b70
+name: Azure DevOps Pull Request Policy Bypassing - Historic Allowlist
+description: |
+  'This detection builds a Allowlist of historic PR policy bypasses and compares to recent history, flagging a non manually allowlisted, non historic pull request bypass.'
+severity: Medium
+queryFrequency: 2h
+queryPeriod: 14d
+triggerOperator: gt
+triggerThreshold: 0
+tactics:
+  - Execution
+  - Impact
+  - Escalation
+query: |
+
+  // Add full UPN (user@domain.com) to Authorized Bypassers to ignore policy bypasses by certain authorized users
+  let AuthorizedBypassers = datatable(UPN:string)[
+  'foo@baz.com'
+  ];
+  let historicBypassers = AzureDevOpsAuditing
+  | where 
+    OperationName == 'Git.RefUpdatePoliciesBypassed' and 
+    TimeGenerated > ago(14d) and TimeGenerated < ago(3h) 
```

Comment:
generally we specify these as parameters at the top - let start = 14d; let end = 3h; and then reference the parameter here.

Reply:
fixed in next commit